DelvEdit/src/com/interrupt/dungeoneer/editor/EditorApplication.java

```diff
@@ -433,18 +434,63 @@ public void loadMonsters() {
 		}
 	}
 
+	/** Creates an empty level with given `width` and `height`. */
 	public void createEmptyLevel(int width, int height) {
 		level = new Level(width, height);
+
+		Tile t = new Tile();
+		t.floorHeight = -0.5f;
+		t.ceilHeight = 0.5f;
+		level.setTile(width / 2, height / 2, t);
+
+		cleanEditorState();
+	}
+
+	/** Generates a level based on a `template` level. */
+	public void generateLevelFromTemplate(Level template) {
+		level.editorMarkers.clear();
+		level.entities.clear();
+		level.non_collidable_entities.clear();
+		level.static_entities.clear();
+
+		level.theme = template.theme;
+		level.generated = true;
+		level.dungeonLevel = 0;
+		level.crop(0, 0, 17 * 5, 17 * 5);
+		level.roomGeneratorChance = 0.4f;
+		level.roomGeneratorType = template.roomGeneratorType;
+		level.generate(Level.Source.EDITOR);
+
+		cleanEditorState();
```

Comment:
I'd prefer not to re-center the camera when generating a level/room. My typical use case is to pan the camera back and mash regenerate to see how the overall generation looks.

Reply:
Sure thing. That sounds like a valid use case :)